sdk/servicebus/service-bus/src/session/messageSession.ts

```diff
@@ -259,105 +254,84 @@ export class MessageSession extends LinkEntity {
     );
   }
 
+  protected createRheaLink(
+    options: ReceiverOptions,
+    _abortSignal?: AbortSignalLike
+  ): Promise<Receiver> {
+    return this._context.connection.createReceiver(options);
+  }
+
   /**
    * Creates a new AMQP receiver under a new AMQP session.
    */
-  private async _init(): Promise<void> {
+  private async _init(abortSignal?: AbortSignalLike): Promise<void> {
     const connectionId = this._context.connectionId;
     try {
-      if (!this.isOpen() && !this.isConnecting) {
-        log.error(
-          "[%s] The receiver '%s' with address '%s' is not open and is not currently " +
-            "establishing itself. Hence let's try to connect.",
-          connectionId,
-          this.name,
-          this.address
-        );
-        this.isConnecting = true;
-        await this._negotiateClaim();
-
-        const options = this._createMessageSessionOptions();
+      const options = this._createMessageSessionOptions();
+      await this.initLink(options, abortSignal);
 
-        log.error(
-          "[%s] Trying to create receiver '%s' with options %O",
-          connectionId,
-          this.name,
-          options
-        );
+      if (this.link == null) {
+        throw new Error("INTERNAL ERROR: failed to create receiver but without an error.");
```

Comment:
Ideally, this never happens, right? Or is there a case?

Reply:
There's no case, that's just to satisfy the type checker.

I could also just '!' it but I figured why not just check it properly.

Reply:
Thought so!